okapi-core/src/main/java/org/folio/okapi/service/impl/ProcessModuleHandle.java

```diff
@@ -125,48 +129,39 @@ private NuProcess launch(Vertx vertx, String id, EnvEntry[] env,
     return process;
   }
 
-  @SuppressWarnings("indentation")
   private Future<Void> start2() {
-    return vertx.executeBlocking(future -> {
-      if (process == null) {
-        String c = "";
-        try {
-          String[] l;
-          if (exec != null) {
-            if (!exec.contains("%p")) {
-              future.fail("Can not deploy: No %p in the exec line");
-              return;
-            }
-            c = exec.replace("%p", Integer.toString(port));
-            l = c.split(" ");
-          } else if (cmdlineStart != null) {
-            if (!cmdlineStart.contains("%p")) {
-              future.fail("Can not deploy: No %p in the cmdlineStart");
-              return;
-            }
-            c = cmdlineStart.replace("%p", Integer.toString(port));
-            l = new String[]{"sh", "-c", c};
-          } else {
-            future.fail("Can not deploy: No exec, no CmdlineStart in LaunchDescriptor");
-            return;
-          }
-          process = launch(vertx, id, env, l);
-          process.waitFor(1, TimeUnit.SECONDS);
-        } catch (InterruptedException ex) {
-          logger.warn("when starting {}", c, ex);
-          Thread.currentThread().interrupt();
-        }
-        if (!process.isRunning() && exitCode != 0) {
-          if (exitCode == Integer.MIN_VALUE) {
-            future.handle(Future.failedFuture(messages.getMessage("11504", c)));
-          } else {
-            future.handle(Future.failedFuture(messages.getMessage("11500", exitCode)));
-          }
-          return;
+    commandLine = "";
+    String[] l;
+    if (exec != null) {
+      if (!exec.contains("%p")) {
+        return Future.failedFuture("Can not deploy: No %p in the exec line");
+      }
+      commandLine = exec.replace("%p", Integer.toString(port));
+      l = commandLine.split(" ");
+    } else if (cmdlineStart != null) {
+      if (!cmdlineStart.contains("%p")) {
+        return Future.failedFuture("Can not deploy: No %p in the cmdlineStart");
+      }
+      commandLine = cmdlineStart.replace("%p", Integer.toString(port));
+      l = new String[]{"sh", "-c", commandLine};
+    } else {
+      return Future.failedFuture("Can not deploy: No exec, no CmdlineStart in LaunchDescriptor");
+    }
+    final String commandLineF = commandLine;
+    process = launch(vertx, id, env, l);
+    Promise<Void> promise = Promise.promise();
+    vertx.setTimer(1000, timerRes -> {
+      if (!process.isRunning() && exitCode != 0) {
```

Comment:
Easier to read:

    if (process.isRunning() || exitCode == 0) {
      promise.complete();
      return;
    }
    if (exitCode == Integer.MIN_VALUE) {
      promise.fail(messages.getMessage("11504", commandLineF));
    } else {
      promise.fail(messages.getMessage("11500", exitCode));
    }

Reply:
ok. In two places.